source/webdav.ts

```diff
@@ -187,31 +187,33 @@ export async function checkConnectivity(): Promise<boolean> {
   return true;
 }
 
-async function rawPipeStream(sanePath: string, stream: Stream): Promise<void> {
+async function rawUpload(saneTargetPath: string, readStream: Stream.Readable): Promise<void> {
   const self: NextcloudClientInterface = this;
 
-  const writeStream = await rawGetWriteStream.call(self, sanePath);
+  const writeStream = await rawGetWriteStream.call(self, saneTargetPath);
 
-  await new Promise((resolve, reject) => {
-    stream.on('error', wrapError);
-    writeStream.on('end', resolve);
-    writeStream.on('error', wrapError);
+  await pipeStreams(readStream, writeStream);
+}
 
-    stream.pipe(writeStream);
+async function rawDownload(saneSourcePath: string, writeStream: Stream.Writable): Promise<void> {
+  const self: NextcloudClientInterface = this;
 
-    function wrapError(error) {
-      reject(NextcloudError(error));
-    }
-  });
+  const readStream = await rawGetReadStream.call(self, saneSourcePath);
+
+  await pipeStreams(readStream, writeStream);
 }
 
-export const createFolderHierarchy = clientFunction(rawCreateFolderHierarchy);
-export const getFolderFileDetails  = clientFunction(rawGetFolderFileDetails);
-export const getFolderProperties   = clientFunction(rawGetFolderProperties);
-export const getWriteStream        = clientFunction(rawGetWriteStream);
-export const getReadStream         = clientFunction(rawGetReadStream);
-export const touchFolder           = clientFunction(rawTouchFolder);
-export const pipeStream            = clientFunction(rawPipeStream);
+export const createFolderHierarchy   = clientFunction(rawCreateFolderHierarchy);
+export const getFolderFileDetails    = clientFunction(rawGetFolderFileDetails);
+export const getFolderProperties     = clientFunction(rawGetFolderProperties);
+export const getWriteStream          = clientFunction(rawGetWriteStream);
+export const getReadStream           = clientFunction(rawGetReadStream);
+export const touchFolder             = clientFunction(rawTouchFolder);
+
+export const pipeStream              = clientFunction(rawUpload); // deprecated
```

Comment:
We should probably emit a warning?! How would one do that in a library ?

Reply:
https://github.com/nodejs/node/issues/11642
https://github.com/dougwilson/nodejs-depd

Reply:
Thought the same. Will do process.emitWarning